File: app/src/test/java/kittentrate/scores/ScoresPresenterTest.java
package kittentrate.scores;

import android.support.v4.app.LoaderManager;

import org.junit.Before;
import org.junit.Test;
import org.mockito.ArgumentCaptor;
import org.mockito.Captor;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

import java.util.List;

import kittentrate.score.PlayerScore;
import kittentrate.score.ScoresContract;
import kittentrate.score.ScoresLoader;
import kittentrate.score.ScoresPresenter;

import static org.mockito.ArgumentMatchers.any;
import static org.mockito.ArgumentMatchers.eq;
import static org.mockito.Mockito.verify;

/**
 * Created by Manuel Lorenzo
 */
public class ScoresPresenterTest {
    @Mock
    private ScoresLoader scoresLoader;
    @Mock
    private LoaderManager loaderManager;
    @Mock
    private ScoresContract.View view;

    private ScoresPresenter scoresPresenter;

    @Before
    public void setUp() throws Exception {
        MockitoAnnotations.initMocks(this);
        scoresPresenter = new ScoresPresenter(view, scoresLoader, loaderManager);
    }

    @Test
    public void testScoresPresenter_loaderManagerShouldInitLoader() throws Exception {
        scoresPresenter.start();

        verify(loaderManager).initLoader(
                eq(ScoresPresenter.LOADER_ID),
                eq(null),
                any(LoaderManager.LoaderCallbacks.class));
    }
}

File: app/src/test/java/kittentrate/scores/ScoresPresenterTest.kt
package kittentrate.scores

import kittentrate.RobolectricTestHelper
import kittentrate.repository.Repository
import kittentrate.ui.score.ScoresContract
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mock
import org.mockito.Mockito.verify
import org.mockito.MockitoAnnotations
import org.robolectric.RobolectricTestRunner

/**
 * Created by Manuel Lorenzo
 */
@RunWith(RobolectricTestRunner::class)
class ScoresPresenterTest : RobolectricTestHelper() {
    @Mock
    private lateinit var view: ScoresContract.View
    @Mock
    private lateinit var repository: Repository

    private lateinit var scoresPresenter: ScoresPresenter

    @Before
    @Throws(Exception::class)
    fun setUp() {
        MockitoAnnotations.initMocks(this)
        scoresPresenter = ScoresPresenter(view, repository)
    }

    @Test
    fun `ScoresPresenter should initialize itself by fetching the top scores from the repo`() {
        scoresPresenter.start()

        verify(repository).topScores
    }
}